megamek/src/megamek/common/actions/WeaponAttackAction.java

```diff
@@ -1780,7 +1780,7 @@ private static String toHitIsImpossible(IGame game, Entity ae, int attackerId, T
                     return Messages.getString("WeaponAttackAction.NoDirectCruiseMissile");
                 }
                 // Direct fire artillery cannot be fired at less than 6 hexes
-                if (isArtilleryDirect && (Compute.effectiveDistance(game, ae, target) <= 6)) {
+                if (isArtilleryDirect && !target.isAirborne() && (Compute.effectiveDistance(game, ae, target) <= 6)) {
                     return Messages.getString("WeaponAttackAction.TooShortForDirectArty");
                 }
```

Comment:
Should this check for Flak ammunition?

Also, I don't believe isAirborne handles landed VTOLs (which do seem to be a valid target based on my read of the rules in question). I'm not even going to ask about the other stuff expending VTOL MPs during a turn they are landed, for example, as that would also be a valid target based on my read of the rules.

Reply:
The arty actually uses standard ammo to do this type of attack.

Per TacOps page 185, the minimum range exception only applies to attacks targeting "airborne aerospace units", so VTOLs still need to be out of minimum range to be targeted.

Reply:
First line states any ordnance with the "F" Damage type in the brackets?

My bad, I missed that bit. Although that raises a different problem, namely isAirborne returns true if the altitude is greater than 0 or the movement mode is either aerodyne or spheroid.

Reply:
I checked it, and MegaMek only allows targeting of airborne aerospace units using F-type ammo (which basically means standard ammo) with direct-fire artillery - cluster and FAE don't show as allowing a direct attack vs airborne ASF. 

I'm not really sure if there is any other problem with using isAirborne(), that's the standard test we use to check if a unit is an ASF and currently in the air. The only point of this specific check is to see if we need to declare the attack as impossible due to being within minimum range.

Reply:
Wouldn't that still return true for a vtol hovering at a higher elevation though?

Reply:
Specifically no. It's only for "airborne units in the manner of an aerospace fighter".

Reply:
There's a separate "isAirborneVTOLorWIGE" method that handles the VTOL situation.